### PR TITLE
BFD-3476: Add feature-flagged C4DIC profile to patient endpoint

### PR DESCRIPTION
**Pull request by aschey-forpeople**
**JIRA Ticket:**
[BFD-3476](https://jira.cms.gov/browse/BFD-3476)

**User Story or Bug Summary:**

---

### What Does This PR Do?

Adds C4DIC profile to the patient endpoint behind a feature flag. This will be disabled by default.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] The data dictionary has been updated with any field mapping changes, if any were made.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/utils/scripts/run-bfd-server [hunk 1]
```diff
@@ -174,7 +174,7 @@ function setvar() {
   else
     full_ssm_name="${SSM_PARAMETER_PATH}/${ssm_name}"
     echo put-parameter $full_ssm_name
-    aws ssm put-parameter --overwrite --endpoint-url http://localhost:4566 --name "${full_ssm_name}" --value "${value}" # > /dev/null
+    aws ssm put-parameter --overwrite --endpoint-url http://localhost:4566 --name "${full_ssm_name}" --value "${value}" --no-cli-pager # > /dev/null
```
Comment (aschey-forpeople) [inline apps/utils/scripts/run-bfd-server:177]: These script changes are unrelated, but having to confirm the SSM parameter updates every time was getting very annoying. Adding `--no-cli-pager` just removes the need to confirm the parameter updates every time it runs.

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/BeneficiaryTransformerV2Test.java [hunk 2]
```diff
@@ -35,18 +35,8 @@
 import java.util.Set;
 import java.util.stream.Collectors;
 import org.hl7.fhir.instance.model.api.IBaseDatatype;
-import org.hl7.fhir.r4.model.Address;
-import org.hl7.fhir.r4.model.BooleanType;
-import org.hl7.fhir.r4.model.Coding;
-import org.hl7.fhir.r4.model.DateTimeType;
-import org.hl7.fhir.r4.model.DateType;
+import org.hl7.fhir.r4.model.*;
```
Comment (aschey-forpeople) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/BeneficiaryTransformerV2Test.java:38]: I can revert this change if needed, something in Intellij keeps insisting on changing it.

Comment (aschey-forpeople) [reply apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/BeneficiaryTransformerV2Test.java]: Found the setting that was doing this, it's fixed now.

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/ProfileConstants.java [hunk 3]
```diff
@@ -53,8 +53,16 @@ public class ProfileConstants {
   /**
    * C4BB Resource Profile: C4BB Patient <a
    * href="http://hl7.org/fhir/us/carin-bb/STU1/StructureDefinition-C4BB-Patient.html">
-   * C4BBOrganization</a>.
+   * C4BBPatient</a>.
```
Comment (aschey-forpeople) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/ProfileConstants.java:56]: Unrelated - this was a pre-existing typo

File: apps/utils/scripts/run-bfd-server (hunk 1 above)
Comment (mjburling) [reply apps/utils/scripts/run-bfd-server:177]: This makes sense and helps avoid the _works on my machine_ commentary. I know on _my machine_, my `~/.aws/config` includes a blanked value for this under default, e.g.

```
[default]
cli_pager=
```

Comment (aschey-forpeople) [reply apps/utils/scripts/run-bfd-server:177]: Ah, makes sense. I didn't know there was a setting for this.

**Review by meliGuzman — APPROVED** [review]

**Review by mjburling — APPROVED** [review]

**Pull request merged**